worker/src/backgroundMigrations/backgroundMigrationManager.ts

```diff
@@ -0,0 +1,178 @@
+import { randomUUID } from "crypto";
+import { IBackgroundMigration } from "./IBackgroundMigration";
+import { prisma, Prisma } from "@langfuse/shared/src/db";
+import { instrumentAsync, logger } from "@langfuse/shared/src/server";
+
+export class BackgroundMigrationManager {
+  private static workerId = randomUUID();
+  private static activeMigration:
+    | {
+        id: string;
+        name: string;
+        args: Record<string, unknown>;
+        migration: IBackgroundMigration;
+      }
+    | undefined;
+
+  private static async heartBeat(): Promise<void> {
+    if (!BackgroundMigrationManager.activeMigration) {
+      return;
+    }
+
+    await prisma.backgroundMigration.updateMany({
+      where: {
+        id: BackgroundMigrationManager.activeMigration.id,
+        workerId: BackgroundMigrationManager.workerId,
+        finishedAt: null,
+        failedAt: null,
+      },
+      data: {
+        lockedAt: new Date(),
+      },
+    });
+
+    // Schedule next heartbeat in 15s
+    setTimeout(BackgroundMigrationManager.heartBeat, 15 * 1000);
+  }
+
+  public static async run(): Promise<void> {
```

Comment:
Thought: maybe we could add prefixes to the logs in this class. This way, self hosters can easily filter logs and share with us.